atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -43,14 +45,62 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def need_skopeo_push(self):
+        if len(self.workflow.exported_image_sequence) > 0:
+            last_image = self.workflow.exported_image_sequence[-1]
+            if last_image['type'] == IMAGE_TYPE_OCI or last_image['type'] == IMAGE_TYPE_OCI_TAR:
+                return True
+
+        return False
+
+    def push_with_skopeo(self, registry_image, insecure, docker_push_secret):
+        # If the last image has type OCI_TAR, then hunt back and find the
+        # the untarred version, since skopeo only supports OCI's as an
+        # untarred directory
+        index = 1
```

Comment:
My version certainly is a bit ugly. It's also wrong - the assertion doesn't work right. Your version looks good. The O(1) version would be something like:

        image = next((x in reversed(self.workflow.exported_image_sequence) if
                      x['type'] != IMAGE_TYPE_OCI_TAR), None)
        assert image is not None

But the use of the next() built-in is a bit cryptic (I had to look it up), and exported_image_sequence is going to be super short.